src/server/req_cred.c

```diff
@@ -0,0 +1,331 @@
+/*
+ * Copyright (C) 1994-2019 Altair Engineering, Inc.
+ * For more information, contact Altair at www.altair.com.
+ *
+ * This file is part of the PBS Professional ("PBS Pro") software.
+ *
+ * Open Source License Information:
+ *
+ * PBS Pro is free software. You can redistribute it and/or modify it under the
+ * terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation, either version 3 of the License, or (at your option) any
+ * later version.
+ *
+ * PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Commercial License Information:
+ *
+ * For a copy of the commercial license terms and conditions,
+ * go to: (http://www.pbspro.com/UserArea/agreement.html)
+ * or contact the Altair Legal Department.
+ *
+ * Altair’s dual-license business model allows companies, individuals, and
+ * organizations to create proprietary derivative works of PBS Pro and
+ * distribute them - whether embedded or bundled with other software -
+ * under a commercial license agreement.
+ *
+ * Use of Altair’s trademarks, including but not limited to "PBS™",
+ * "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+ * trademark licensing policies.
+ *
+ */
+
+/**
+ * @file	req_cred.c
+ *
+ * @brief
+ *  Server routines providing and sending credentials to superior mom.
+ */
+
+#include <pbs_config.h>   /* the master config generated by configure */
+
+#include "libpbs.h"
+#include "job.h"
+#include "batch_request.h"
+#include "pbs_error.h"
+#include "log.h"
+#include "pbs_nodes.h"
+#include "server.h"
+
+#define	CRED_DATA_SIZE	4096
+
+/* credential type */
+#define CRED_NONE	0
+#define CRED_KRB5	1
+
+extern struct server server;
+extern void release_req(struct work_task *);
+extern int relay_to_mom(job *, struct batch_request *, void (*)(struct work_task *));
+extern time_t time_now;
+pbs_list_head svr_creds_cache;	/* all credentials cached and available to send */
+extern long svr_cred_renew_cache_period;
+
+struct cred_cache {
+	pbs_list_link	cr_link;
+	char	  credid[PBS_MAXUSER+1];
```

Comment:
spaces around '+' is needed.
We can also remove extra spaces between type and variable names.